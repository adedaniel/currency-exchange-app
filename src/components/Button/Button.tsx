import React, { FC } from "react";

interface IButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
}

const Button: FC<IButtonProps> = (props) => {
  const { className, ...rest } = props;
  return (
    <button
      className={`w-full h-12 border-[1px] rounded-md ${className}`}
      data-testid="button"
      {...rest}
    />
  );
};

export default Button;
