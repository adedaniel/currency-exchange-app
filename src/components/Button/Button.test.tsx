import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Button from "./Button";

describe("Button", () => {
  it("renders without crashing", () => {
    const wrapper = render(<Button />);

    expect(wrapper).toBeTruthy();
  });
});

it("renders button text correctly", () => {
  const { getByTestId } = render(<Button children="test" />);

  expect(getByTestId("button")).toHaveTextContent("test");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button children="test" />).toJSON();

  expect(tree).toMatchSnapshot();
});
