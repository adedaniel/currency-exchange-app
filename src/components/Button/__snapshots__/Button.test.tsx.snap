// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="w-full h-12 border-[1px] rounded-md undefined"
  data-testid="button"
>
  test
</button>
`;
